qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -379,24 +412,35 @@ def _assemble_noise_model(self, circuits, **run_options):
         noise_model = run_options.get(
             'noise_model', getattr(self.options, 'noise_model', None))
 
-        # Add custom pass noise only to QuantumCircuit objects
-        if noise_model and all(isinstance(circ, QuantumCircuit) for circ in circuits):
+        # Add custom pass noise only to QuantumCircuit objects that contain delay
+        # instructions since this is the only instruction handled by the noise pass
+        # at present
```

Comment:
It would perhaps be better to add a function to `noise_model` that takes (for example) the circuit and the opset, and returns whether it would affect it - it feels like `AerBackend` shouldn't need to know this for all models, because different noise models might do different things.

Reply:
Probably worth doing something like this eventually